No images available for release v3.8.0

Spot checked a few of these images on https://hub.docker.com/u/stackstorm and didn't find a release tag for v3.8.0.

Example from the st2web image tags section:

The most recent release tag was `3.7.0`. Following dev images available:

- 3.8dev
- 3.9dev 

Also looks like `latest` was published 7 mo ago (probably at v3.7.0)

![image](https://user-images.githubusercontent.com/29960411/204680371-3ec80e4b-29d6-47f3-bd55-1ea9f65dd287.png)

Thanks for reporting.
Per https://app.circleci.com/pipelines/github/StackStorm/st2-dockerfiles/1171/workflows/f788c9e4-ee29-47a6-8546-ef87e8f8272a/jobs/1704 looks like the build failed for the v3.8 branch:

```
#!/bin/sh -eo pipefail
make build

Failed to identify the tags to be set.
No images were tagged due to an error when determining the correct tags: Error: Unexpected HTTP statuscode for https://registry.hub.docker.com/v1/repositories/stackstorm/st2/tags: 410
exit 1
make: *** [Makefile:101: verify_tag_update_flag] Error 1

Exited with code exit status 2


```

--- a/determine_needed_tags.sh
+++ b/determine_needed_tags.sh
@@ -78,20 +78,20 @@ tag_update_flag=0
 #     or if the build is for a completely new st2 component)
 
 # check if there are already images for the given component available at dockerhub
-dockerhub_registry_status_code=$(curl -sfL -o /dev/null -w "%{http_code}" https://registry.hub.docker.com/v1/repositories/stackstorm/${component}/tags)
+dockerhub_registry_status_code=$(curl -sfL -o /dev/null -w "%{http_code}" https://hub.docker.com/v2/repositories/stackstorm/${component}/tags)
 
 if [ ${dockerhub_registry_status_code} -eq 404 ]; then
   # there is no repository stackstorm/${component} available at dockerhub -> this build is for a new st2 component
   tag_update_flag=3
   exit
 elif [ ${dockerhub_registry_status_code} -ne 200 ]; then
-  echo "Error: Unexpected HTTP statuscode for https://registry.hub.docker.com/v1/repositories/stackstorm/${component}/tags: ${dockerhub_registry_status_code}"
+  echo "Error: Unexpected HTTP statuscode for https://hub.docker.com/v2/repositories/stackstorm/${component}/tags: ${dockerhub_registry_status_code}"
   exit 1
 fi
 
 # dockerhub lists the tags in ascending order. 1st object = lowest tag; last object = highest tag or latest
-docker_tags_json=$(curl -s https://registry.hub.docker.com/v1/repositories/stackstorm/${component}/tags)
-readarray -t available_releases < <(echo $docker_tags_json | jq -r '.[] | select((.name | endswith("dev") | not) and (.name=="latest" | not)).name' | sort)
+docker_tags_json=$(curl -s https://hub.docker.com/v2/repositories/stackstorm/${component}/tags)
+readarray -t available_releases < <(echo $docker_tags_json | jq -r '.results | .[] | select((.name | endswith("dev") | not) and (.name=="latest" | not)).name' | sort)
 
 # sort returns the list with the highest tag or "latest" as last item
 # so change the value below to -2 when introducing the tag latest to get i.e. 3.3.0 instead of latest
@@ -113,7 +113,7 @@ else
   # building a release for a st2 version that does not match the latest version
   if [ ${build_major} -le ${latest_major} ]; then
     # building a release for an older major
-    readarray -t build_version_major_minor_matching_releases < <(echo $docker_tags_json | jq -r '.[] | select(.name | endswith("dev") | not) | select(.name | startswith("'"${build_major}.${build_minor}"'")).name')
+    readarray -t build_version_major_minor_matching_releases < <(echo $docker_tags_json | jq -r '.results | .[] | select(.name | endswith("dev") | not) | select(.name | startswith("'"${build_major}.${build_minor}"'")).name')
     if [ ${#build_version_major_minor_matching_releases[@]} -ge 1 ]; then
       # at least one version matching the current builds major and minor version is available
       latest_build_version_major_minor_matching_release=${build_version_major_minor_matching_releases[-1]}
@@ -129,7 +129,7 @@ else
       elif [ ${build_minor} -eq ${latest_build_version_major_minor_matching_minor} ] && \
            [ ${build_patch} -ge ${latest_build_version_major_minor_matching_patch} ]; then
         # building a release for a new or updated patch version of an old major.minor version
-        readarray -t build_version_major_matching_releases < <(echo $docker_tags_json | jq -r '.[] | select(.name | endswith("dev") | not) | select(.name | startswith("'"${build_major}"'")).name')
+        readarray -t build_version_major_matching_releases < <(echo $docker_tags_json | jq -r '.results | .[] | select(.name | endswith("dev") | not) | select(.name | startswith("'"${build_major}"'")).name')
         latest_build_version_major_matching_release=${build_version_major_matching_releases[-1]}
         latest_build_version_major_matching_release_array=(${latest_build_version_major_matching_release//\./ })
         latest_build_version_major_matching_minor=${latest_build_version_major_matching_release_array[1]}

